clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1343,9 +1343,15 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
 
     if (YcArg || YuArg) {
       StringRef ThroughHeader = YcArg ? YcArg->getValue() : YuArg->getValue();
-      // Enable PCH inclusion when performing host compilation with -fsycl.
-      if (!isa<PrecompileJobAction>(JA) && JA.isOffloading(Action::OFK_SYCL) &&
-          !JA.isDeviceOffloading(Action::OFK_SYCL)) {
+      // If PCH file is available, include it while performing
+      // host compilation (-fsycl-is-host) in SYCL mode (-fsycl).
+      // as well as in non-sycl mode.
+      bool NonSYCLCompilation = !JA.isOffloading(Action::OFK_SYCL);
+      bool SYCLHostCompilation = JA.isOffloading(Action::OFK_SYCL) &&
+                                 !JA.isDeviceOffloading(Action::OFK_SYCL);
```

Comment:
The enabling of PCH consumption is basically only restricted for SYCL device compilations.  Why not just do a `bool SYCLDeviceCompilation = JA.isOffloading(Action::OFK_SYCL) && JA.isDeviceOffloading(Action::OFK_SYCL);` and just check for `!SYCLDeviceCompilation` when adding the appropriate options?